api/core/v1alpha1/vtep_types.go

```diff
@@ -0,0 +1,163 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// VTEPSpec defines the desired state of a VXLAN Tunnel Endoint (VTEP)
+// +kubebuilder:validation:XValidation:rule="self.anycastInterfaceRef.name != self.primaryInterfaceRef.name",message="anycastInterfaceRef.name must differ from primaryInterfaceRef.name"
+type VTEPSpec struct {
+	// DeviceName is the name of the Device this object belongs to. The Device object must exist in the same namespace.
+	// Immutable.
+	// +required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="DeviceRef is immutable"
+	DeviceRef LocalObjectReference `json:"deviceRef"`
+
+	// ProviderConfigRef is a reference to a resource holding the provider-specific configuration for this VTEP.
+	// +optional
+	ProviderConfigRef *TypedLocalObjectReference `json:"providerConfigRef,omitempty"`
+
+	// Enable indicates whether the interface is administratively up or down.
+	// +required
+	Enabled bool `json:"enabled"`
+
+	// PrimaryInterfaceRef is the reference to the loopback interface used for the primary VTEP IP address.
+	// +required
+	PrimaryInterfaceRef LocalObjectReference `json:"primaryInterfaceRef"`
+
+	// AnycastInterfaceRef is the reference to the loopback interface used for anycast VTEP IP address.
+	// +required
+	AnycastInterfaceRef LocalObjectReference `json:"anycastInterfaceRef,omitempty"`
```

Comment:
```suggestion
	// +optional
	AnycastInterfaceRef *LocalObjectReference `json:"anycastInterfaceRef,omitempty"`
```
This should actually be optional.